api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -171,6 +171,26 @@ message HttpConnectionManager {
   // :ref:`config_http_conn_man_headers_x-forwarded-for` for more information.
   uint32 xff_num_trusted_hops = 19;
 
+  // If set, Envoy will not append the remote address to the
+  // :ref:`config_http_conn_man_headers_x-forwarded-for` HTTP header. This may be used in
+  // conjunction with HTTP filters that explicitly manipulate XFF after the HTTP connection manager
+  // has mutated the request headers. While :ref:`use_remote_address
+  // <config_http_conn_man_use_remote_address>` will also suppress XFF addition, it has consequences
+  // for logging and other Envoy uses of the remote address, so *skip_xff_append* should be used
+  // when only an elision of XFF addition is intended.
+  bool skip_xff_append = 21;
+
+  // Never consider requests to be of internal provenance when performing :ref:`header sanitization
+  // <config_http_conn_man_header_sanitizing>`. Envoy will scrub various *x-envoy-* headers that are
+  // only allowed to be set by internal network elements.
+  bool external_only = 22;
```

Comment:
nit: This is a pretty vague name. Thoughts on something longer and more descriptive? `consider_all_requests_external` is a mouthful but clearer. Something like that?

Reply:
Actually, for this specific one, due to some technicalities of how we do networking (logical vs. physical addressing), I think we get what we are after effectively for free today, so I'm going to skip it.